syft/execution/plan.py

```diff
@@ -396,41 +504,67 @@ def __call__(self, *args, **kwargs):
           and use the result(s) to instantiate to appropriate placeholder.
         - Return the instantiation of all the output placeholders.
         """
-        if self.forward is not None:  # if not self.is_built:
+
+        def instantiate_input(input_placeholders, args):
+            """
+                Function to instantiate all the placeholders in order, even if in nested structures.
+
+                Args:
+                    input_placeholders: the input placeholders generated at build time.
+                    args: the arguments received on __call__
+            """
+
+            for arg in args:
+                if isinstance(arg, FrameworkTensor):
+                    placeholder = input_placeholders.pop(0)
+                    placeholder.instantiate(arg)
+                elif isinstance(arg, (list, tuple, set)):
+                    instantiate_input(input_placeholders, arg)
+                elif isinstance(arg, dict):
+                    instantiate_input(input_placeholders, list(arg.values()))
```

Comment:
for dict, is there a risk that the value() don't  come in the expected order? (dict is not ordered)

Reply:
Nice observation, I'll add this to the TODO list and I will think about this, currently this is might be bug prone.